CMakeLists.txt

```diff
@@ -10,6 +10,9 @@ add_definitions(-DLOG_USE_COLOR)
 set(CMAKE_VERBOSE_MAKEFILE ON)
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -g -pthread -fPIC -Werror")
 
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} --std=c++14 -fpermissive -Wall -Wextra -lpthread")
```

Comment:
Be aware that C++14 may not fully supported by Visual Studio 2015: 
https://msdn.microsoft.com/en-us/library/hh409293.aspx

Make sure the code can be compiled on Windows.